From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH] ia64: kprobes section conflict

 This patch moves jprobe_return from being a static inline that does
 nothing to a function that does nothing. The reason is that jprobe_return
 is exported, so it causes a section type conflict during build.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
---

 asm-ia64/kprobes.h    |    4 +---
 ia64/kernel/kprobes.c |    2 ++
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/include/asm-ia64/kprobes.h	2007-06-12 15:45:40.000000000 -0400
+++ b/include/asm-ia64/kprobes.h	2007-06-15 13:01:48.000000000 -0400
@@ -125,9 +125,7 @@
 				    unsigned long val, void *data);
 
 /* ia64 does not need this */
-static inline void jprobe_return(void)
-{
-}
+void jprobe_return(void);
 extern void invalidate_stacked_regs(void);
 extern void flush_register_stack(void);
 extern void arch_remove_kprobe(struct kprobe *p);
--- a/arch/ia64/kernel/kprobes.c	2007-06-15 13:03:10.000000000 -0400
+++ b/arch/ia64/kernel/kprobes.c	2007-06-15 13:03:11.000000000 -0400
@@ -36,6 +36,8 @@
 #include <asm/uaccess.h>
 
 extern void jprobe_inst_return(void);
+/* ia64 doesn't need this, but the symbol is exported */
+void jprobe_return(void) {}
 
 DEFINE_PER_CPU(struct kprobe *, current_kprobe) = NULL;
 DEFINE_PER_CPU(struct kprobe_ctlblk, kprobe_ctlblk);
